pypeit/core/flat.py

```diff
@@ -90,7 +83,7 @@ def tweak_slit_edges(slit_left_in, slit_righ_in, ximg_fit, normimg, tweak_slits_
     return slit_left_out, slit_righ_out, tweak_dict
 
 
-def fit_flat(flat, tilts_dict, tslits_dict_in, slit, slitmask_func = Spectrograph.slitmask, inmask = None,spec_samp_fine = 1.2, spec_samp_coarse = 50.0,
+def fit_flat(flat, tilts_dict, tslits_dict_in, slit, spectrograph = None, inmask = None,spec_samp_fine = 1.2, spec_samp_coarse = 50.0,
```

Comment:
Yeah look I'd prefer to not pass in spectrograph. I  passed in the slit mask function before and you complained, so I changed to this, and now you are complaining again;). The point is that this routine tweaks the slits, and in order to tweak the slits it needs to know how to reconstruct the slit mask with masking of partial orders. So either it takes the objects or it takes the function. I preferred the function because that is a standard thing and many python routines take functions as arguments (like all the optimization ones), but you complained.